corehq/apps/domain/tests/test_delete_domain.py

```diff
@@ -1100,5 +1103,39 @@ def test_hard_delete_cases_none_to_delete(self):
 
 
 def ensure_deleted(domain):
-    if domain._rev:
-        domain.delete()
+    def make_exe(op):
+        """Make execute function that ignores failed SQL deletions and continue
+        deleting other related entities, especially Couch objects.
+
+        SQL entities that cannot be deleted will be cleaned up on
+        rollback at the end of the transaction.
+        """
+        def execute(domain_name):
+            try:
+                real_execute(domain_name)
+            except TransactionManagementError as err:
+                if cant_execute not in str(err):
+                    raise
+        real_execute = op.execute
+        return execute
+
+    cant_execute = "can't execute queries until the end of the 'atomic' block."
```

Comment:
Might be a nit, but does this need to be defined at this scope? Or can it be defined within `execute`?

Reply:
Could be defined in either. I put it here since it's referenced in a loop that is more deeply nested and indentation made the line extra long. Also no need to allocate it repeatedly for each iteration (although Python probably optimizes that anyway since it's an immutable string).